File: Top 100 Codes/02-Working with Numbers/13-Permutations in which n people can occupy r seats in a classroom/Permutationsinwhichnpeoplecanoccupyrseatsinaclassroom.cpp
// Problem Statement :
// In a classroom some of the seats are already occupied by students and only a few seats are available in the classroom. The available seats are assumed as r and n number of students are looking for the seat. We need to find in how many different permutations n number of students can sit on r number of chairs.
#include <iostream>
using namespace std;

// Function to calculate factorial of a number
int factorial(int num) {
    int fact = 1;
    for (int i = num; i >= 1; i--) {
        fact *= i; // Multiply the current value with the factorial
    }
    return fact; // Return the computed factorial
}

int main() {
    int n, r;

    // Prompt the user for the number of people
    cout << "Enter the number of people (n): ";
    cin >> n;

    // Prompt the user for the number of seats
    cout << "Enter the number of seats (r): ";
    cin >> r;

    // Check if the inputs are valid (n >= r)
    if (n < r) {
        cout << "Error: Number of people (n) must be greater than or equal to the number of seats (r)." << endl;
        return 1; // Exit the program with an error code
    }

    // Calculate the total number of arrangements using the permutation formula
    // P(n, r) = n! / (n - r)!
    int permutations = factorial(n) / factorial(n - r);

    // Display the result
    cout << "Total possible arrangements: " << permutations << endl;

    return 0; // Indicate successful execution
}

File: Top 100 Codes/02-Working with Numbers/13-Permutations in which n people can occupy r seats in a classroom/Permutationsinwhichnpeoplecanoccupyrseatsinaclassroom.py
# Function to calculate factorial of a number
def factorial(num):
    fact = 1
    for i in range(num, 1, -1):  # Loop from num down to 2
        fact *= i  # Multiply each number to calculate factorial
    return fact  # Return the calculated factorial


# Main program
# User input for the number of people (n) and seats (r)
n = int(input("Enter the number of people (n): "))
r = int(input("Enter the number of seats (r): "))

# Check if the input is valid (n >= r)
if n < r:
    print("Error: Number of people (n) must be greater than or equal to the number of seats (r).")
else:
    # Calculate permutations using the formula P(n, r) = n! / (n - r)!
    permutations = factorial(n) // factorial(n - r)

    # Print the result
    print("Total possible arrangements:", permutations)
